### PR TITLE
Refactor profile screen

### LINKED ISSUE #247: Refactor Profile screens
In our current state, the profile screens use a lot of common code that could be used in common within the `profile` package. Work should be done to modularize and refactor this package.

### PR DESCRIPTION
**Pull request by matttsss**
### What is this PR?
This PR refactors the `profile` package, "extracts" and modularises the common components used in the `EditProfileScreen` and `ViewProfileScreen`.

### Why?
This is done in order to reduce boiler plate code when editing profiles and thus could be re-used for the later PR of creating an account screen.  

### Test coverage:
image

Closes #247

### CODE REVIEW

File: app/src/main/java/com/github/se/cyrcle/ui/profile/EditProfileComponent.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            onValueChange = { username = it },
+            label = stringResource(R.string.view_profile_screen_username_label),
+            minCharacters = USERNAME_MIN_LENGTH,
+            maxCharacters = USERNAME_MAX_LENGTH,
+            testTag = "UsernameField")
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
+          saveButton(
```
Comment (sean-prz) [inline app/src/main/java/com/github/se/cyrcle/ui/profile/EditProfileComponent.kt:100]: If you do another commit on this branch before merging, can you please change the order of the save and cancel button, it's really not intuitive and I keep pressing the wrong one. If not I'll take care of it when doing the profile picture

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt [hunk 2]
```diff
@@ -387,7 +206,7 @@ private fun FavoriteParkingCard(parking: Parking, index: Int, onRemove: () -> Un
               onClick = {
                 onRemove()
                 showConfirmDialog = false
-                Toast.makeText(context, "Removed from Favorites!", Toast.LENGTH_SHORT).show()
+                Toast.makeText(context, onFavParkingRemovalText, Toast.LENGTH_SHORT).show()
```
Comment (Armetiis) [inline app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:209]: This is quite unrelated to your PR per se, but I think we could drop this toast. The user can see that the favorite parking is gone, thus making this toast a bit overkill

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt [hunk 3]
```diff
@@ -62,18 +60,17 @@ fun ViewProfileScreen(
...
-  var lastName by remember { mutableStateOf(userState?.details?.lastName ?: "") }
-  var username by remember { mutableStateOf(userState?.public?.username ?: "") }
-  var profilePictureUrl by remember { mutableStateOf(userState?.public?.profilePictureUrl ?: "") }
-  val context = LocalContext.current
 
-  val imagePickerLauncher =
-      rememberLauncherForActivityResult(ActivityResultContracts.GetContent()) { uri ->
-        uri?.let { profilePictureUrl = it.toString() }
-      }
+  val signOutToastTest = stringResource(R.string.view_profile_on_sign_out_toast)
```
Comment (Armetiis) [inline app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt:73]: Here you wrote "Test" instead of "Text"

**Review by Armetiis — APPROVED** [review]
You did a good job modularizing the code. It's much cleaner and maintanable now! I have nothing to add, well done

File: app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt (hunk 2 above)
Comment (matttsss) [reply app/src/main/java/com/github/se/cyrcle/ui/profile/ViewProfileScreen.kt]: Thanks for the suggestion! Code coverage :arrow_up: :fire:

**Review by sean-prz — APPROVED** [review]

**Review by Armetiis — APPROVED** [review]

**Pull request merged**